app.js

```diff
@@ -121,14 +134,45 @@ router.post('/contract/view/:name/:methodName', async ctx => {
 
 router.get('/account/:name', async ctx => {
     const response = await client.request('view_account', [{
-        account_id: await hash(ctx.params.name),
+        account_id: await encodeAccountNameForRpc(ctx.params.name),
         method_name: '',
         args: []
     }]);
     checkError(ctx, response);
     ctx.body = response.result;
 });
 
+/**
+ * Create a new account. Generate a throw away account id (UUID).
+ */
+router.post('/account', async ctx => {
+    // TODO: this is using alice account to create all accounts. We may want to change that.
+    const nonce = await getNonce(ctx, defaultSender);
+    ensureTrue(nonce);
+    const newAccountName = uuidV4();
+    console.log("Creating new account " + newAccountName);
+
+    // TODO: unhardcode key
+    const createAccountParams = {
+        'nonce': nonce,
+        'sender': await encodeAccountNameForRpc(defaultSender),
+        'new_account_id': await encodeAccountNameForRpc(newAccountName),
+        'amount': newAccountAmount,
+        'public_key': hardcodedKey.public_key,
+    };
+
+    const resp = await submit_transaction_rpc(client, "create_account", createAccountParams);
+    checkError(ctx, resp);
+    ctx.body = resp.result;
+});
+
+/*
+ * Helper to encode the account name in the format which is used in rpcs.
+ */
+async function encodeAccountNameForRpc(plainTextName){
```

Comment:
this function doesn't do anything. If you think `hash` isn't clear enough name – just rename it.

Reply:
I wanted to have a wrapper so that, if we decide to stop hashing names as per conversation yesterday, we don't have to change in 10 places. hash is a good name for readability, so I want to keep that as a separate thing. this said, I don't care enough to insist on this.

Reply:
we don't use `hash` for anything else. We can rename it to something like `accountHash` to be more specific, I'm ok with that.